src/App.ts

```diff
@@ -719,7 +719,38 @@ export default class App {
       };
     };
 
-    // Set body and payload (this value will eventually conform to AnyMiddlewareArgs)
+    // Set body and payload
+    // TODO: this value should eventually conform to AnyMiddlewareArgs
+    let payload: any = {};
+    switch (type) {
+      case IncomingEventType.Event:
+        payload = (bodyArg as SlackEventMiddlewareArgs['body']).event;
+        break;
+      case IncomingEventType.ViewAction:
+        payload = (bodyArg as SlackViewMiddlewareArgs['body']).view;
+        break;
+      case IncomingEventType.Shortcut:
+        payload = (bodyArg as SlackShortcutMiddlewareArgs['body']);
+        break;
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore: Fallthrough case in switch
```

Comment:
We have [`noFallthroughCasesInSwitch`](https://github.com/slackapi/bolt-js/blob/tslint-to-eslint/tsconfig.json#L37) enabled in `tsconfig.json`, which I think is a sensible default. However, I think this is one situation where a fallthrough in a case in a switch actually works to preserve the logic.

What does everyone think? Too complicated? Too many linter / compiler ignore comments to go along with it?

Reply:
>// If above conditional does not hit, fall through to fallback payload in default block below

This is reasonable enough, so that I'm happy to ignore the rule here.